include/mbedtls/ecdh.h

```diff
@@ -70,7 +116,26 @@ typedef struct mbedtls_ecdh_context
 #if defined(MBEDTLS_ECP_RESTARTABLE)
     int restart_enabled;        /*!< The flag for restartable mode. */
     mbedtls_ecp_restart_ctx rs; /*!< The restart context for EC computations. */
-#endif
+#endif /* MBEDTLS_ECP_RESTARTABLE */
+#else
+    uint8_t point_format;       /*!< The format of point export in TLS messages
+                                  as defined in RFC 4492. */
+    mbedtls_ecp_group_id grp_id;/*!< The elliptic curve used. */
+    mbedtls_ecdh_variant var;   /*!< The ECDH implementation/structure used. */
+    union
+    {
+        mbedtls_ecdh_context_mbed   mbed_ecdh;
+    } ctx;                      /*!< Implementation-specific context. The
```

Comment:
The comment here is confusing. Also, this is a union of what? I only see `mbed_ecdh`

Reply:
I have just seen that PR. This means that every alternative implementation should modify `ecdh.h` to add their context?
I don't agree with this design. I think it is better to have some `_ALT` context , that every alternative implementation should define in their own specific `alt` header file .
I understand this is a bit late in the design process, but I strongly suggest rethinking this.

Reply:
The goal of this mechanism different from the ALT one. Here we expect a small number of alternatives that go through us, not independent overriding of software implementations by a hardware driver. And we want to be able to link multiple alternatives.